diff --git a/gdk/meson.build b/gdk/meson.build
index 6b2f50950d..84c98fe2e2 100644
--- a/gdk/meson.build
+++ b/gdk/meson.build
@@ -246,6 +246,11 @@ libgdk_c_args = [
   '-DG_LOG_DOMAIN="Gdk"',
 ]

+gdk_static_compilation_args = []
+if get_option('default_library') == 'static'
+  gdk_static_compilation_args += ['-DGDK_STATIC_COMPILATION']
+endif
+
 gdk_backends = []
 gdk_backends_gen_headers = []  # non-public generated headers
 foreach backend : ['broadway', 'wayland', 'win32', 'x11', 'macos']
@@ -270,7 +275,7 @@ libgdk = static_library('gdk',
   dependencies: gdk_deps + [libgtk_css_dep],
   link_with: [libgtk_css],
   include_directories: [confinc, gdkx11_inc, wlinc],
-  c_args: libgdk_c_args + common_cflags,
+  c_args: libgdk_c_args + common_cflags + gdk_static_compilation_args,
   link_whole: gdk_backends,
 )

@@ -281,4 +286,5 @@ libgdk_dep = declare_dependency(
   sources: ['gdk.h', gdkconfig, gdkenum_h],
   include_directories: [confinc, gdkx11_inc, wlinc],
   dependencies: gdk_deps + [libgtk_css_dep],
+  compile_args: gdk_static_compilation_args,
 )
